### PR TITLE
Display POST body parameters as tables

### PR DESCRIPTION
**Pull request by richvdh**
Replace a whole bunch of special-casing for POST body parameters with the same
logic as is used for response objects: represent all but the top-level as
tables.

Once this, #255, *and* #254 all land, we'll have documentation of the params to POST /filter.

### CODE REVIEW

File: templating/matrix_templates/units.py [hunk 1]
```diff
@@ -548,6 +460,32 @@ def _load_swagger_meta(self, filepath, api, group_name):
...
+            req_body_tables = get_tables_for_schema(filepath, param["schema"])
+        except Exception, e:
+            logger.warning("Error decoding body of API endpoint %s %s: %s",
+                           endpoint_data["method"], endpoint_data["path"],
+                           e.args[0])
+            return
+
+        # put the top-level parameters into 'req_param_by_loc', and the others
+        # into 'req_body_tables'
+        body_params = endpoint_data['req_param_by_loc'].setdefault("JSON body",[])
```
Comment (illicitonion) [inline templating/matrix_templates/units.py]: nit: s/'/"/g and on line 486

**Comment by illicitonion** [comment]
LGTM

**Comment by illicitonion** [comment]
Actually, I *believe* this isn't **required**-ing all of the params, e.g. search RoomEvent in http://matrix.org/speculator/spec/257/client_server.html

**Comment by richvdh** [comment]
Hrm. That's not new, but I'll look into it.

**Comment by richvdh** [comment]
Wait. RoomEvent only appears in *responses*. It doesn't seem to make 
sense to mark response fields as "required".

**Pull request merged**